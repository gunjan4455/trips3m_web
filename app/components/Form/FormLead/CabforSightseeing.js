import React from 'react';
import PropTypes from 'prop-types';

import {
  CabIconFront
} from 'helpers/Icon/Icon';

const CabforSightseeing = ({ input, question, cabQuestionHandler }) => {
  const isChecked = (optionId) => {
    try {
      let index = -1;
      if (input.value !== '') {
        index = [input.value].findIndex(obj =>
          obj.option_id.toString() === optionId.toString());
      }
      return (index !== -1);
    } catch (e) {
      return false;
    }
  };

  const handleChange = (event) => {
    if (!question.useInputForOptions) {
      input.onChange({
        option_id: event.target.value.toString(),
        customization_id: question.customizationId.toString()
      });
    } else {
      // TODO :: set user input
      input.onChange({
        user_input: '',
        option_id: event.target.value.toString(),
        customization_id: question.customizationId.toString()
      });
    }
    cabQuestionHandler(event.target.value);
  };

  return (
    <div>
      <div className="flight-included col-xs-8 pr0 pl0 at_cab">
        <p className="fw9 relative m0"><span className="flight-icon"><CabIconFront /></span>
          {question.title}
        </p>
      </div>
      <div className="col-xs-4 pr0 pl0 text-right at_cab_radio">
        {
          question.options.map((option, index) =>
            <div className="col-xs-6 pl0 pr0" key={option.id}>
              <input
                type="radio"
                className="flight-radio"
                name={`${question.identifier}[${index}]`}
                id={`radio_button_${option.id}_${index}`}
                value={option.id}
                checked={isChecked(option.id)}
                onChange={handleChange}
              />
              <label htmlFor={`radio_button_${option.id}_${index}`} className="cursorP pr8 pl8">
                <p className="m0 pfc4 f12 fw4 text-uppercase">{option.name}</p>
              </label>
            </div>)
        }
      </div>
    </div>
  );
};

CabforSightseeing.propTypes = {
  question: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
  cabQuestionHandler: PropTypes.func
};

CabforSightseeing.defaultProps = {
  cabQuestionHandler: () => {}
};
export default CabforSightseeing;
