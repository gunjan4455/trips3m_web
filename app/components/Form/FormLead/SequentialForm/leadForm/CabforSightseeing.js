import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { CabsFilled, Tick } from 'helpers/Icon/Icon';

class CabforSightseeing extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedIndex: 0,
      isChecked: this.selectCab(props),
    };
  }

  selectCab = (props) => {
    const { formValues } = props;
    if(formValues.other_answers_attributes && formValues.other_answers_attributes.length) {
      return formValues.other_answers_attributes[1] &&
        formValues.other_answers_attributes[1].option_id &&
        formValues.other_answers_attributes[1].option_id === "99";
    }
    return false;
  };

  handleChange = event => {
    const { input, question } = this.props;
    this.setState({ isChecked: !this.state.isChecked }, () => {
      input.onChange({
        option_id: question.options[this.state.isChecked ? 0 : 1].id,
        customization_id: question.customizationId.toString(),
      });
    });
  };

  render() {
    const { question } = this.props;
    const { isChecked } = this.state;
    const optionIndex = isChecked ? 0 : 1;
    const option = question.options[optionIndex];
    return (
      <div className={`button-box round-icon ${isChecked ? 'active' : ''}`}>
        <span className="tick-icon-container">
          <Tick/>
        </span>
        <div className="icon-48">
          <CabsFilled />
        </div>
        <input
          type="checkbox"
          className="checkbox-common"
          name={`${question.identifier}[${optionIndex}]`}
          id={`radio_button_${option.id}_${optionIndex}`}
          value={option.id}
          checked={isChecked}
          onChange={this.handleChange}
        />
        <p className="text-center">Cab</p>
      </div>
    );
  }
}

CabforSightseeing.propTypes = {
  question: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
  formValues: PropTypes.object.isRequired,
  cabQuestionHandler: PropTypes.func,
};

CabforSightseeing.defaultProps = {
  cabQuestionHandler: () => {},
};
export default CabforSightseeing;
