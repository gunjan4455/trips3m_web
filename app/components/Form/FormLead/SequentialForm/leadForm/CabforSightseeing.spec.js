import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import CabforSightseeing from './CabforSightseeing';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  input: {
    name: 'other_answers_attributes[1]',
    value: '',
    onChange: () => {
    }
  },
  meta: {
    active: false,
    asyncValidating: false,
    autofilled: false,
    dirty: false,
    form: 'LeadForm',
    invalid: false,
    pristine: true,
    submitting: false,
    submitFailed: false,
    touched: false,
    valid: true,
    visited: false
  },
  question: {
    customizationId: 24,
    customizationTags: 'cab',
    description: '',
    hasOthers: false,
    id: 136,
    identifier: 'cust_24',
    jsonValueType: 'hash',
    options: [
      {
        id: '99',
        misc: '',
        name: 'Yes',
        tags: '',
        url: ''
      },
      {
        id: '100',
        misc: '',
        name: 'No',
        tags: '',
        url: ''
      }
    ],
    questionSequence: 1,
    railsFormName: 'other_answers_attributes',
    templateId: '1',
    templateType: 'radio_button',
    title: 'Cab for local sightseeing?',
    useInputForOptions: false,
    valueType: 'single_value'
  },
  formValues: {
    email: null,
    dont_utilize_budget_new: true,
    budget_new: true,
    start_date_noidea: false,
    adult: '2',
    child: 0,
    isd_code: '+91',
    child_age_answers_attributes: [
      {
        user_input: '0',
        customization_id: '7',
        option_id: '29'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '30'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '31'
      }
    ],
    syncErrors: {
      email: 'Enter valid email id'
    },
    to_loc: 'Kerala',
    start_date: '2019-07-17',
    start_date_flexible: false,
    trip_days: '2',
    hotel_needed_answer_attributes: {
      option_id: '95',
      customization_id: '22'
    },
    hotel_category_answers_attributes: [
      {
        option_id: '7',
        customization_id: '2'
      }
    ]
  }
}

const shallowWrapperCabforSightseeing = (props) => {
  return shallow(<CabforSightseeing {...props} />);
};

describe('lead form module', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperCabforSightseeing(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange properly', () => {
    const wrapper = shallowWrapperCabforSightseeing(props);
    wrapper.instance().handleChange({});
    expect(wrapper.state().isChecked).toEqual(true);
  });

  it('should isChecked in state true when other_answers_attributes are available', () => {
    props.formValues.other_answers_attributes = [
      {
        option_id: '7',
        customization_id: '2'
      },
      {
        option_id: '99',
        customization_id: '2'
      }
    ];
    const wrapper = shallowWrapperCabforSightseeing(props);
    expect(wrapper.state().isChecked).toEqual(true);
  });

});
