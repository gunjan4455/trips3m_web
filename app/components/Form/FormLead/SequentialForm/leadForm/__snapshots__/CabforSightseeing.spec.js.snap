// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lead form module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CabforSightseeing
    cabQuestionHandler={[Function]}
    formValues={
      Object {
        "adult": "2",
        "budget_new": true,
        "child": 0,
        "child_age_answers_attributes": Array [
          Object {
            "customization_id": "7",
            "option_id": "29",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "30",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "31",
            "user_input": "0",
          },
        ],
        "dont_utilize_budget_new": true,
        "email": null,
        "hotel_category_answers_attributes": Array [
          Object {
            "customization_id": "2",
            "option_id": "7",
          },
        ],
        "hotel_needed_answer_attributes": Object {
          "customization_id": "22",
          "option_id": "95",
        },
        "isd_code": "+91",
        "start_date": "2019-07-17",
        "start_date_flexible": false,
        "start_date_noidea": false,
        "syncErrors": Object {
          "email": "Enter valid email id",
        },
        "to_loc": "Kerala",
        "trip_days": "2",
      }
    }
    input={
      Object {
        "name": "other_answers_attributes[1]",
        "onChange": [Function],
        "value": "",
      }
    }
    meta={
      Object {
        "active": false,
        "asyncValidating": false,
        "autofilled": false,
        "dirty": false,
        "form": "LeadForm",
        "invalid": false,
        "pristine": true,
        "submitFailed": false,
        "submitting": false,
        "touched": false,
        "valid": true,
        "visited": false,
      }
    }
    question={
      Object {
        "customizationId": 24,
        "customizationTags": "cab",
        "description": "",
        "hasOthers": false,
        "id": 136,
        "identifier": "cust_24",
        "jsonValueType": "hash",
        "options": Array [
          Object {
            "id": "99",
            "misc": "",
            "name": "Yes",
            "tags": "",
            "url": "",
          },
          Object {
            "id": "100",
            "misc": "",
            "name": "No",
            "tags": "",
            "url": "",
          },
        ],
        "questionSequence": 1,
        "railsFormName": "other_answers_attributes",
        "templateId": "1",
        "templateType": "radio_button",
        "title": "Cab for local sightseeing?",
        "useInputForOptions": false,
        "valueType": "single_value",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="tick-icon-container"
        >
          <IconSvg />
        </span>,
        <div
          className="icon-48"
        >
          <IconSvg />
        </div>,
        <input
          checked={false}
          className="checkbox-common"
          id="radio_button_100_1"
          name="cust_24[1]"
          onChange={[Function]}
          type="checkbox"
          value="100"
        />,
        <p
          className="text-center"
        >
          Cab
        </p>,
      ],
      "className": "button-box round-icon ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "tick-icon-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "icon-48",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "className": "checkbox-common",
          "id": "radio_button_100_1",
          "name": "cust_24[1]",
          "onChange": [Function],
          "type": "checkbox",
          "value": "100",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cab",
          "className": "text-center",
        },
        "ref": null,
        "rendered": "Cab",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="tick-icon-container"
          >
            <IconSvg />
          </span>,
          <div
            className="icon-48"
          >
            <IconSvg />
          </div>,
          <input
            checked={false}
            className="checkbox-common"
            id="radio_button_100_1"
            name="cust_24[1]"
            onChange={[Function]}
            type="checkbox"
            value="100"
          />,
          <p
            className="text-center"
          >
            Cab
          </p>,
        ],
        "className": "button-box round-icon ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "tick-icon-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "icon-48",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "checkbox-common",
            "id": "radio_button_100_1",
            "name": "cust_24[1]",
            "onChange": [Function],
            "type": "checkbox",
            "value": "100",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cab",
            "className": "text-center",
          },
          "ref": null,
          "rendered": "Cab",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
